addons/account/models/account_move.py

```diff
@@ -4703,277 +4703,336 @@ def _mail_track(self, tracked_fields, initial):
-            skip_reconciliation = False
-            if recon_currency.is_zero(recon_debit_amount):
-                debit_line = None
-                skip_reconciliation = True
-            if recon_currency.is_zero(recon_credit_amount):
-                credit_line = None
-                skip_reconciliation = True
-            if skip_reconciliation:
-                continue
-
-            # ==== Match both lines together and compute amounts to reconcile ====
+                partial_debit_amount_currency = 0.0
+            if credit_rate:
+                partial_credit_amount_currency = credit_vals['currency'].round(credit_rate * min_recon_amount)
+                partial_credit_amount_currency = min(partial_credit_amount_currency, -remaining_credit_amount_curr)
+            else:
+                partial_credit_amount_currency = 0.0
 
-            # Determine which line is fully matched by the other.
-            compare_amounts = recon_currency.compare_amounts(recon_debit_amount, recon_credit_amount)
-            min_recon_amount = min(recon_debit_amount, recon_credit_amount)
-            debit_fully_matched = compare_amounts <= 0
-            credit_fully_matched = compare_amounts >= 0
+        else:
+            # recon_currency != company_currency
+            # Compute the partial amount expressed in company currency.
+            if debit_rate:
+                partial_debit_amount = company_currency.round(min_recon_amount / debit_rate)
+                partial_debit_amount = min(partial_debit_amount, remaining_debit_amount)
+            else:
+                partial_debit_amount = 0.0
+            if credit_rate:
+                partial_credit_amount = company_currency.round(min_recon_amount / credit_rate)
+                partial_credit_amount = min(partial_credit_amount, -remaining_credit_amount)
+            else:
+                partial_credit_amount = 0.0
+            partial_amount = min(partial_debit_amount, partial_credit_amount)
+
+            # Compute the partial amount expressed in foreign currency.
+            # Take care to handle the case when a line expressed in company currency is mimicking the foreign
+            # currency of the opposite line.
+            if debit_vals['currency'] == company_currency:
+                partial_debit_amount_currency = partial_amount
+            else:
+                partial_debit_amount_currency = min_recon_amount
+            if credit_vals['currency'] == company_currency:
+                partial_credit_amount_currency = partial_amount
+            else:
+                partial_credit_amount_currency = min_recon_amount
 
-            # ==== Computation of partial amounts ====
+        # Computation of the partial exchange difference. You can skip this part using the
+        # `no_exchange_difference` context key (when reconciling an exchange difference for example).
+        if not self._context.get('no_exchange_difference'):
```

Comment:
Can't we reverse the key? It's a negative one: `no_exchange_difference=True` is a bit counter-intuitive.
Changing it to `compute_exchange_difference` is clearer to me. (In the current state, it makes me think of the `exclude_from_invoice_tab` which is sometime confusing)